src/math_tools/partial_derivatives.hpp

```diff
@@ -0,0 +1,73 @@
+
+// SPDX-License-Identifier: MIT
+/**
+ * @file partial_derivatives.hpp
+ * File containing functions to compute the partial derivatives
+ */
+
+#pragma once
+
+#include <ddc/ddc.hpp>
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+
+
+/**
+ * @brief Compute the partial derivative of a 
+ *
+ * For a given field @f$F(X1,X2)@f$ , compute
+ * @f$\partial_{Xi} F(X1,X2)@f$ with i=1 or 2.
+ *
+ * @param[in] Xi
+ *      The given coordinate for the partial derivative.
+ *
+ * @return @f$\partial_{Xi} F(X1,X2)@f$ 
+ */
+
+template <
+        class GridXi,
+        class GridX1,
+        class GridX2,
+        class FieldX1Builder,
+        class FieldX1Evaluator>
+class PartialDerivative
```

Comment:
`GridXi` is unused. It should either be removed or it should be used for static asserts